src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -735,20 +735,29 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
       "--"
    else
       "#"
-   
+
    reCommentPrefix <- .rs.reRmdInlineChunk(commentPrefix)
    for (line in code[-1]) {
-      
-      # there may be variable number of yaml chunk opts, but they should all be at the top of the chunk
+      # there may be variable number of in-body chunk opts,
+      # but they should all be at the top of the chunk
       match <- unlist(regmatches(line, regexec(reCommentPrefix, line)))
-      if (length(match) == 0)
+      if (length(match) == 0 && commentPrefix != "#")
+      {
+         # if the language-specific comment prefix character was not used,
+         # fallback and check for starting with #| instead before bailing
+         match <- unlist(regmatches(line, regexec(.rs.reRmdInlineChunk("#"), line)))
+         if (length(match) == 0)
+            break
+         # if we found a match, we also need to update what gets passed to .rs.parseInlineChunk
+         commentPrefix <- "#"
+         reCommentPrefix <- .rs.reRmdInlineChunk(commentPrefix)
+      }
+      else if (length(match) == 0)
```

Comment:
basically, the behavior in https://github.com/yihui/knitr/blob/607148df836902b8d1c2f496df97f2f83159cac4/R/parser.R#L279-L282

if the language-specific comment prefix is not `#`, but the user used `#` anyway, we update `commentPrefix` and `reCommentPrefix` accordingly and add the match to inline chunk options. We don't really allow a mix of `#|` and `--|` in a single chunk, but I don't think we should (and I doubt anyone is mixing and matching within a chunk).